File: app/src/main/java/com/donald/musictheoryapp/Music/MusicXML/Part.java
package com.donald.musictheoryapp.Music.MusicXML;

public class Part
{
	public final String id;
	public final Measure[] measures;

	public Part(String id, Measure[] measures)
	{
		this.id = id;
		this.measures = measures;
	}

	public Measure[] measures()
	{
		return measures;
	}

	public Part clone()
	{
		Measure[] otherMeasures = new Measure[measures.length];
		for(int i = 0; i < otherMeasures.length; i++)
		{
			otherMeasures[i] = measures[i].clone();
		}
		return new Part(id, otherMeasures);
	}
}

File: app/src/main/java/com/donald/musictheoryapp/music/MusicXML/Part.kt
package com.donald.musictheoryapp.music.MusicXML

import org.dom4j.Element

class Part(val id: String, val measures: Array<Measure>) {

    fun measures(): Array<Measure> {
        return measures
    }

    fun clone(): Part {
        val otherMeasures = Array(measures.size) { index ->
            measures[index].clone()
        }
        return Part(id, otherMeasures)
    }

    fun addToXml(scorePartwise: Element) {
        scorePartwise.addElement("part").apply {
            addAttribute("id", id)
            for (measure in measures) {
                measure.addToXml(this)
            }
        }
    }

}